components/camel-smpp/src/main/java/org/apache/camel/component/smpp/AbstractSmppCommand.java

```diff
@@ -96,34 +98,37 @@ protected Message getResponseMessage(Exchange exchange) {
      * @return
      */
     @Deprecated
-    @SuppressWarnings("rawtypes")
+    @SuppressWarnings({ "rawtypes", "unchecked" })
     protected List<OptionalParameter> createOptionalParametersByName(Map<String, String> optinalParamaters) {
         List<OptionalParameter> optParams = new ArrayList<OptionalParameter>();
 
         for (Entry<String, String> entry : optinalParamaters.entrySet()) {
             OptionalParameter optParam = null;
 
+            String value = entry.getValue();
             try {
                 Tag tag = Tag.valueOf(entry.getKey());
                 Class type = determineTypeClass(tag);
 
-                if (OctetString.class.equals(type)) {
-                    optParam = new OptionalParameter.OctetString(tag.code(), entry.getValue());
-                } else if (COctetString.class.equals(type)) {
-                    optParam = new OptionalParameter.COctetString(tag.code(), entry.getValue());
-                } else if (org.jsmpp.bean.OptionalParameter.Byte.class.equals(type)) {
-                    optParam = new OptionalParameter.Byte(tag.code(), Byte.valueOf(entry.getValue()));
-                } else if (org.jsmpp.bean.OptionalParameter.Int.class.equals(type)) {
-                    optParam = new OptionalParameter.Int(tag.code(), Integer.valueOf(entry.getValue()));
-                } else if (org.jsmpp.bean.OptionalParameter.Short.class.equals(type)) {
-                    optParam = new OptionalParameter.Short(tag.code(), Short.valueOf(entry.getValue()));
-                } else if (org.jsmpp.bean.OptionalParameter.Null.class.equals(type)) {
-                    optParam = new OptionalParameter.Null(tag);
+                Class superclass = type.getSuperclass();
+                Class supersuperclass = superclass.getSuperclass();
```

Comment:
As this method only works with `OptionalParameter` this should not be the case. I will update the pull request anyway.